crates/agglayer-grpc-api/src/certificate_submission_service/mod.rs

```diff
@@ -67,3 +76,57 @@ where
         }))
     }
 }
+
+pub(crate) fn get_extra_signature(
+    metadata: &tonic::metadata::MetadataMap,
+) -> Result<Option<Signature>, Box<Status>> {
+    if let Some(value) = metadata.get(GRPC_METADATA_NAME_EXTRA_CERTIFICATE_SIGNATURE) {
+        let sig_str = value.to_str().map_err(|e| {
+            tonic::Status::invalid_argument(format!("invalid signature encoding: {e}"))
+        })?;
+
+        let sig = Signature::from_str(sig_str).map_err(|e| {
+            tonic::Status::invalid_argument(format!("invalid hex in signature: {e}"))
+        })?;
+
+        Ok(Some(sig))
+    } else {
+        Ok(None)
+    }
```

Comment:
Depending on the taste really (feel free to ignore suggestion), but alternative would be:

```suggestion
    metadata
        .get(GRPC_METADATA_NAME_EXTRA_CERTIFICATE_SIGNATURE)
        .map(|value| {
            value
                .to_str()
                .map_err(|e| {
                    Box::new(tonic::Status::invalid_argument(format!(
                        "invalid signature encoding: {e}"
                    )))
                })
                .and_then(|sig_str| {
                    Signature::from_str(sig_str).map_err(|e| {
                        Box::new(tonic::Status::invalid_argument(format!(
                            "invalid hex in signature: {e}"
                        )))
                    })
                })
        })
        .transpose()
```